### ISSUE TITLE
Use the abc package for abstract Python classes

### ISSUE THREAD

**Issue by danthedaniel** [c-mcg/File-Server#10]
Python allows for abstract classes that define methods which must be implemented (with tools located in the `abc` package). This might be useful for the `Packet` class.

**Comment by c-mcg** [comment]
Absolutely, thanks. This should also be used in `HubProcessor` and `file_server/web/endpoints/`

https://docs.python.org/3.6/library/abc.html
